### DISCUSSION TITLE
Not possible to run SvelteKit SPA inside a subfolder of a web server

### DISCUSSION THREAD

**Discussion by GELight** [sveltejs/kit#2837] [Q&A]
### Describe the bug

Its not possible to run a SvelteKit SAP application in a **subfolder of a web server** like this:

**Folder example:**
wwwroot/project_folder

Url example:
http://mydomain.de/project_folder

### Reproduction

1. Create new sveltekit app with `npm init svelte@next my-app`
2. Add adapter `@sveltejs/adapter-static` to the project and the sveltekit configuration
3. Add the path configuration for the `config.kit.paths`
```
path: {
   base: '/project_folder',
   assets: 'http://localhost/project_folder'
}
```
4. Run `npm run build` to create a production version of the project
5. Create an example web server root folder out of the project `mkdir wwwroot`
6. Copy the created production `my-app/build` folder into the `wwwroot`
7. Rename `wwwroot/build` folder to `my-app/project_folder`
8. Go into `wwwroot` folder and start a web server by `http-server`
9. Open URL http://localhost/project_folder
10. Click any link to route to other sub route

**Problem is:**
The router will not change the pages per Javascript. Instead a standard behaviour of the browser with a default page refresh (no javascript) from `/` to `/blog` (for example) happens. The result is a 404 error because the page does not exist.

### Logs

```shell
No errors in log...
```

### System Info

```shell
Tested in Linux Mint...
Tested in Windows...
```

### Severity

blocking all usage of SvelteKit

### Additional Information

**Question from my side:**

1. Is there any configuration to running my SAP inside a sub folder in my web server?
2. Its possible to configure the router for using `hash` in URL like in [Vue.js - hash-mode](https://next.router.vuejs.org/guide/essentials/history-mode.html#hash-mode)?
3. Maybe its needed any server rewrite rule configuration for the project folder `wwwroot/project_folder`?

The problem is ... only one webserver ( IIS ) is possible. 
Therefore only working with sub folders is possible in our company.

**Comment by dominikg** [comment]
Please use discussions and/or discord https://svelte.dev/chat for questions

**Comment by edwinspire** [comment]
I had the same problem with my project with Sveltekit and adapter-static when uploading it on my IIS where I have several apps in other folders.

What I did is place in svelte.config.js:
```

import adapter from '@sveltejs/adapter-static';
import { vitePreprocess } from '@sveltejs/kit/vite';

const config = {
	preprocess: vitePreprocess(),

	kit: {
		adapter: adapter({
			fallback: 'index.html',
		}),
		paths: {base: '/svelte'},
		prerender: { entries: [] }
	}
};

export default config;
```


Where the base "/svelte" is the name of the folder where I saved the files generated by Sveltekit en IIS. 

![image](https://user-images.githubusercontent.com/3129446/218633623-f6b11b8c-8724-4340-8391-532b6a457cdb.png)

![image](https://user-images.githubusercontent.com/3129446/218633930-301967aa-4549-49ce-90ea-a1ccd3d21708.png)


It worked.
